src/sidebar/helpers/annotation-metadata.ts

```diff
@@ -208,12 +208,12 @@ export function isHidden(annotation: Annotation): boolean {
  * Highlights are generally identifiable by having no text content AND no tags,
  * but there is some nuance.
  */
-export function isHighlight(annotation: Annotation): boolean {
+export function isHighlight(annotation: APIAnnotationData): boolean {
   // `$highlight` is an ephemeral attribute set by the `annotator` on new
   // annotation objects (created by clicking the "highlight" button).
   // It is not persisted and cannot be relied upon, but if it IS present,
   // this is definitely a highlight (one which is not yet saved).
-  if (annotation.$highlight) {
+  if ('$highlight' in annotation && annotation.$highlight) {
```

Comment:
The `$highlight` property doesn't exist in `APIAnnotationData`, so TS can't reason about it here. What you could do is make the argument `Annotation | APIAnnotationData` instead.

Reply:
Good idea.